src/businesslogic/BusinessLogic.java

```diff
@@ -1,12 +1,37 @@
 package businesslogic;
 
 
+import products.Product;
+
 public class BusinessLogic {
 
 
+    @Deprecated
     public static double calculateIva(double price) {
         int iva = 22;
         return (price * iva) / 100;
     }
 
+    public double getCalculateDiscount23(Product product) {
+        DiscountPercentage percentage= DiscountPercentage.PERCENTAGE23;
+        double priceDiscount = (Integer.parseInt(percentage.getPercentage()) * product.getPrice()) / 100;
+        double result = product.getPrice() - priceDiscount;
+        return result;
+    }
+
+
+    public double getCalculateDiscount9(Product product) {
+        DiscountPercentage percentage= DiscountPercentage.PERCENTAGE9;
+        double priceDiscount = (Integer.parseInt(percentage.getPercentage()) * product.getPrice()) / 100;
```

Comment:
come parametro il metodo deve prendere anche la percentuale di sconto così da poter applicare in ogni prodotto uno sconto diverso

Reply:
non mi convince neanche il parse int

Reply:
rispondo al primo messaggio: va bene.
rispondo al secondo messaggio: su internet spiega in questa maniera, funziona perfettamente, prende la stringa e la casta in intero.